src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -45,5 +47,1075 @@ class Outer
+                // private file class E { } // 3, 4
+                Diagnostic(ErrorCode.ERR_FileTypeNoExplicitAccessibility, "E").WithArguments("E").WithLocation(3, 20),
+                // (3,20): error CS1527: Elements defined in a namespace cannot be explicitly declared as private, protected, protected internal, or private protected
+                // private file class E { } // 3, 4
+                Diagnostic(ErrorCode.ERR_NoNamespacePrivate, "E").WithLocation(3, 20));
+        }
+
+        [Fact]
+        public void BaseClause_01()
+        {
+            var source = """
+                file class Base { }
+                class Derived1 : Base { } // 1
+                public class Derived2 : Base { } // 2, 3
+                file class Derived3 : Base { } // ok
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (2,7): error CS9301: File type 'Base' cannot be used as a base type of non-file type 'Derived1'.
+                // class Derived1 : Base { } // 1
+                Diagnostic(ErrorCode.ERR_FileTypeBase, "Derived1").WithArguments("Base", "Derived1").WithLocation(2, 7),
+                // (3,14): error CS0060: Inconsistent accessibility: base class 'Base' is less accessible than class 'Derived2'
+                // public class Derived2 : Base { } // 2, 3
+                Diagnostic(ErrorCode.ERR_BadVisBaseClass, "Derived2").WithArguments("Derived2", "Base").WithLocation(3, 14),
+                // (3,14): error CS9301: File type 'Base' cannot be used as a base type of non-file type 'Derived2'.
+                // public class Derived2 : Base { } // 2, 3
+                Diagnostic(ErrorCode.ERR_FileTypeBase, "Derived2").WithArguments("Base", "Derived2").WithLocation(3, 14));
+        }
+
+        [Fact]
+        public void BaseClause_02()
+        {
+            var source = """
+                file interface Interface { }
+
+                class Derived1 : Interface { } // ok
+                file class Derived2 : Interface { } // ok
+
+                interface Derived3 : Interface { } // 1
+                file interface Derived4 : Interface { } // ok
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (6,11): error CS9301: File type 'Interface' cannot be used as a base type of non-file type 'Derived3'.
+                // interface Derived3 : Interface { } // 1
+                Diagnostic(ErrorCode.ERR_FileTypeBase, "Derived3").WithArguments("Interface", "Derived3").WithLocation(6, 11));
+        }
+
+        [Fact]
+        public void TypeArguments_01()
+        {
+            var source = """
+                file class C { }
+                class Container<T> { }
+                class Program
+                {
+                    Container<C> M() => new Container<C>(); // 1
```

Comment:
nit; Along those lines, consider expanding to usages in array types, tuple types, pointer types, function pointer types, ...

Reply:
Also haven't gotten to this yet.

Reply:
Done